Celeste.Mod.mm/Patches/FakeHeart.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.Xna.Framework;
+using Monocle;
+using MonoMod;
+
+namespace Celeste {
+    public class patch_FakeHeart : FakeHeart {
+        public patch_FakeHeart(Vector2 position)
+            : base(position) {
+            // dummy constructor
+        }
+
+        // null is the vanilla (random) color.
+        private AreaMode? color;
+
+        [MonoModConstructor]
+        [MonoModIgnore]
+        public extern void ctor(Vector2 position);
+
+        [MonoModConstructor]
+        [MonoModReplace]
+        public void ctor(EntityData data, Vector2 offset) {
+            ctor(data.Position + offset);
+
+            if (data.Has("color") && data.Attr("color") != "Random") {
+                color = data.Enum<AreaMode>("color");
+            } else {
+                color = null;
+            }
```

Comment:
`EntityData.Enum` is case-insensitive, so the `Random` check should be as well.

```suggestion
            if (data.Has("color") && data.Attr("color").ToLower() != "random") {
                color = data.Enum<AreaMode>("color");
            } else {
                color = null;
            }
```

As discussed on Discord, it'd be better to default to random on invalid color values anyway.

```suggestion
            color = data.Enum<AreaMode>("color", (AreaMode) (-1));
```